# HG changeset patch
# User Michael Tharp <mtharp@rpath.com>
# Date 1281538755 14400
# Node ID 65d49b270bb131bda773d12bd6230eda60b788c9
# Parent  b45bcd8a59154f048b7a425d6b97bcaf15ae27cc
Fix an import bug that caused weird interactions with the lazy importer. (CNY-3490)

diff --git a/NEWS.src/CNY-3490.bugfix b/NEWS.src/CNY-3490.bugfix
new file mode 100644
--- /dev/null
+++ b/NEWS.src/CNY-3490.bugfix
@@ -0,0 +1,2 @@
+A bug introduced in Conary 2.1.21 that prevented `cvc cook` from functioning
+has been corrected. (CNY-3490)
diff --git a/conary/build/cook.py b/conary/build/cook.py
--- a/conary/build/cook.py
+++ b/conary/build/cook.py
@@ -40,6 +40,7 @@
 from conary.lib import debugger, log, logger, sha1helper, util, magic
 from conary.local import database
 from conary.repository import changeset, errors, filecontents
+from conary.conaryclient import callbacks as client_callbacks
 from conary.conaryclient.cmdline import parseTroveSpec
 from conary.state import ConaryState, ConaryStateFromFile
 
@@ -156,7 +157,7 @@
 
 # -------------------- public below this line -------------------------
 
-class CookCallback(conaryclient.callbacks.ChangesetCallback, callbacks.CookCallback):
+class CookCallback(client_callbacks.ChangesetCallback, callbacks.CookCallback):
 
     def buildingChangeset(self):
         self._message('Building changeset...')
